programs/degen-pools/src/pools.rs

```diff
@@ -87,6 +89,45 @@ pub fn set_winning_option(ctx: Context<UpdatePool>, winning_option: Pubkey) -> R
     Ok(())
 }
 
+pub fn fund_pool<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, FundPool<'info>>,
+    value: u64,
+) -> Result<()> {
+    let pool_account = &mut ctx.accounts.pool_account;
+    if pool_account.is_paused {
+        return err!(CustomError::PoolStateIncompatible);
+    }
+
+    if ctx.remaining_accounts.len() == 0 {
+        return err!(CustomError::NoPoolOptionAccounts);
+    }
+
+    let remaining_accounts = ctx.remaining_accounts;
+    let value_per_option = value / remaining_accounts.len() as u64;
+    for account in remaining_accounts {
+        let mut pool_option = match Account::<PoolOption>::try_from(account) {
+            Ok(pool_option) => pool_option,
+            Err(_) => return err!(CustomError::InvalidPoolOptionAccount),
+        };
+        pool_option.value += value_per_option;
+        pool_option.exit(&PROGRAM_ID)?;
+    }
+
+    // Transfer SOL from caller to pool account
+    let cpi_context = CpiContext::new(
```

Comment:
is there a way to use the existing `enter_pool` fn in `entries.rs`?

That will also take away the need to duplicate this check: https://github.com/Degen-Markets/degen-markets-solana/pull/12/files#diff-e4fe02d9ea4178a1c23a95acc10402633301397989504c69116d932a115bfefeR97

Reply:
Checking now. Thanks for telling, I missed this check